raft: minimize/strengthen TestLogMaybeAppend

pav-kv opened the pull request:
`TestLogMaybeAppend` was testing both `maybeAppend` and `commitTo` methods. The latter is already tested in `TestCommitTo`, so we don't need to test both and have combinatorial explosion in `TestLogMaybeAppend`.

This PR minimizes the test, and makes the checks in it stronger (verify the entire content of the log instead of just a sub-slice).

This is a prerequisite for further clean-ups around `maybeAppend`, `commitTo` and the corresponding tests.

Related to #126320
Epic: none
Release note: none

pav-kv committed:
raft: clean up TestLogMaybeAppend

Epic: none
Release note: none

pav-kv committed:
raft: clean up commitTo test and commenting

Epic: none
Release note: none

pav-kv committed:
raft: minimize TestLogMaybeAppend

TestLogMaybeAppend was testing both maybeAppend and commitTo method. The
latter is already tested in TestCommitTo, so we don't need to test both
and have combinatorial explosion in TestLogMaybeAppend.

Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/126407)
<!-- Reviewable:end -->

pav-kv commented:
Commit-by-commit review is recommended.

pav-kv committed:
raft: make TestLogMaybeAppend stronger

The test was only verifying that a sub-slice of the log matches the
appended slice. This commit, instead, makes it verify the entire content
of the log.

Epic: none
Release note: none

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo nits

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-:-O0VG5cAERm-NRhd0Yqz:b-1p9zor)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @pav-kv)

___
*[`pkg/raft/log_test.go` line 207 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFMCU1Rm81PsTnbZS:-O0VFMCU1Rm81PsTnbZT:b-bnqch4) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca37d10adb333b662965f242b8d1e2fb4a290697/pkg/raft/log_test.go#L207)):*
> ```Go
> }
> 
> // TestLogMaybeAppend ensures:
> ```

nit: should we update the comment instead of removing it entirely? 
___
*[`pkg/raft/log_test.go` line 218 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFFvz0JzZ0jWekyvB:-O0VFFvz0JzZ0jWekyvC:bqx09lo) ([raw file](https://github.com/cockroachdb/cockroach/blob/1c9e5ba1a8579e82ac815ef435626a93a41e7b3f/pkg/raft/log_test.go#L218)):*
> ```Go
> 		ents []pb.Entry
> 
> 		wappend bool
> ```

nit: want to rename these while you're here as well? Maybe expAppend and expPanic?

---

EDIT: no longer relevant as of r3.
___
*[`pkg/raft/log_test.go` line 215 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFljD4iQs58FO49qg:-O0VFljD4iQs58FO49qh:by4kh9i) ([raw file](https://github.com/cockroachdb/cockroach/blob/35ea0db3f79dbadf56d336795720b777c837e2e2/pkg/raft/log_test.go#L215)):*
> ```Go
> 	for _, tt := range []struct {
> 		app   logSlice
> 		notOk bool
> ```

nit: consider calling this `expOk` instead. That way, you don't need the negation in the assertion below.
___
*[`pkg/raft/log_test.go` line 215 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFy_ZAMKoRGvpPSR9:-O0VFy_ZAMKoRGvpPSRA:bed0u8z) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe85b55875df8e727a49be347c1967e3a119d1b6/pkg/raft/log_test.go#L215)):*
> ```Go
> 	for _, tt := range []struct {
> 		app   logSlice
> 		want  []pb.Entry
> ```

nit: s/want/expEntries?


<!-- Sent from Reviewable.io -->

pav-kv reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/126407)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani)

___
*[`pkg/raft/log_test.go` line 207 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFMCU1Rm81PsTnbZS:-O0VJmGU5kacV62TyIY4:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca37d10adb333b662965f242b8d1e2fb4a290697/pkg/raft/log_test.go#L207)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: should we update the comment instead of removing it entirely?
</blockquote></details>

Done
___
*[`pkg/raft/log_test.go` line 215 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFljD4iQs58FO49qg:-O0VGNc8967NdLH8rXo6:b-da5gnz) ([raw file](https://github.com/cockroachdb/cockroach/blob/35ea0db3f79dbadf56d336795720b777c837e2e2/pkg/raft/log_test.go#L215)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: consider calling this `expOk` instead. That way, you don't need the negation in the assertion below.
</blockquote></details>

The negation is intentional, because most of the tests are about the ok (the default) case. This eliminates boilerplate, consider this a "pattern". This is also consistent with the `panic` variable: by default there is no panic, but otherwise we type `panic: true` explicitly.
___
*[`pkg/raft/log_test.go` line 215 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/126407#-O0VFy_ZAMKoRGvpPSR9:-O0VI5um3wsgg7GYa6Ma:b-vsump3) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe85b55875df8e727a49be347c1967e3a119d1b6/pkg/raft/log_test.go#L215)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: s/want/expEntries?
</blockquote></details>

Leaving as is, for brevity. When there is only one "want"/"expected" kind of assertion, I prefer short names, as the semantics of the check is obvious from the code.


<!-- Sent from Reviewable.io -->
